components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -136,6 +148,24 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         return true;
     }
 
+    @Override
+    public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        // In the Asgardeo console's token system, users are given two tokens upon login - an Authorization code token
+        // and an Organization Switch Grant token. The latter is generated using the former and is utilized to
+        // access resources. Upon logout, only the Authorization code token is revoked, leaving the
+        // Organization Switch Grant token active.  To address this, a prefix is added to the Authorization code token's
+        // binding reference, which is then used in the Organization Switch Grant token. Therefore, during logout,
+        // all tokens associated with the binding reference, including those with the prefix, are revoked,
+        // ensuring both tokens are effectively revoked.
+        String tokenBindingRef = "os_" + tokReqMsgCtx.getProperty(
```

Comment:
Shall we extract the "os_" prefix to a constant?

Reply:
The changes made by the merged PR have been rolled back. The associated issue is now being resolved with a different PR and is being monitored under the issue wso2-enterprise/asgardeo-product#17656.